docs/usage/design.rst

```diff
@@ -0,0 +1,26 @@
+Design of SCAutolib
+=============================
+
+This section would introduce design decisions that were made during
```

Comment:
I would write:
'describes' instead of 'would introduce'; 
'their explanation'  instead of 'explanation why is it decided in that particular way'